frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.unit.spec.tsx

```diff
@@ -0,0 +1,234 @@
+import { Route } from "react-router";
+
+import { renderWithProviders, screen } from "__support__/ui";
+import {
+  createMockSettingsState,
+  createMockState,
+} from "metabase-types/store/mocks";
+import {
+  createMockDatabase,
+  createMockTokenFeatures,
+  createMockUser,
+} from "metabase-types/api/mocks";
+import type { TokenFeatures } from "metabase-types/api";
+import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+
+import {
+  setupDatabaseEndpoints,
+  setupDatabasesEndpoints,
+} from "__support__/server-mocks";
+
+import { DatabasePromptBanner } from "./DatabasePromptBanner";
+
+interface SetupOpts {
+  isAdmin?: boolean;
+  isPaidPlan?: boolean;
+  onlyHaveSampleDatabase?: boolean;
+  isOnAdminAddDatabasePage?: boolean;
+}
+const TEST_DB = createSampleDatabase();
+
+const DATA_WAREHOUSE_DB = createMockDatabase({ id: 2 });
+
+function setup({
+  isAdmin = false,
+  isPaidPlan = false,
+  onlyHaveSampleDatabase = false,
+  isOnAdminAddDatabasePage = false,
+}: SetupOpts = {}) {
+  setupDatabaseEndpoints;
+  if (onlyHaveSampleDatabase) {
+    setupDatabasesEndpoints([TEST_DB]);
+  } else {
+    setupDatabasesEndpoints([TEST_DB, DATA_WAREHOUSE_DB]);
+  }
+
+  const state = createMockState({
+    currentUser: createMockUser({ is_superuser: isAdmin }),
+    settings: createMockSettingsState({
+      "token-features": createMockTokenFeatures(
+        isPaidPlan ? randomizePaidPlanFeatures() : {},
+      ),
+    }),
+  });
+
+  renderWithProviders(
+    isOnAdminAddDatabasePage ? (
+      <Route path="/admin/databases/create" component={DatabasePromptBanner} />
+    ) : (
+      <Route path="*" component={DatabasePromptBanner} />
+    ),
+    {
+      initialRoute: isOnAdminAddDatabasePage ? "/admin/databases/create" : "/",
+      storeInitialState: state,
+      withRouter: true,
+    },
+  );
+}
+
+describe("DatabasePromptBanner", () => {
+  it("should not render for non-admin users without paid plan without connected databases", () => {
+    setup({
```

Comment:
In contract to the previous comment, I really like these tests from the `setup` point of view - if you're an admin, on a paid plan, and have other databases and not just the sample one, you shouldn't see the paid banner. Makes sense from the product point of view.